crates/broker/src/order_picker.rs

```diff
@@ -464,10 +510,37 @@ where
     /// Estimate of gas for locking a single order
     /// Currently just uses the config estimate but this may change in the future
     async fn estimate_gas_to_lock(&self, _order: &Order) -> Result<u64> {
+        // TODO: Add gas costs for orders with smart contract signatures.
         Ok(self.config.lock_all().context("Failed to read config")?.market.lockin_gas_estimate)
     }
 
+    /// Estimate of gas for to fulfill a single order
+    /// Currently just uses the config estimate but this may change in the future
+    async fn estimate_gas_to_fulfill(&self, order: &Order) -> Result<u64> {
+        // TODO: Add gas costs for orders with large journals.
+        let (base, groth16) = {
+            let config = self.config.lock_all().context("Failed to read config")?;
+            (config.market.fulfill_gas_estimate, config.market.groth16_verify_gas_estimate)
+        };
+
+        match self
+            .supported_selectors
+            .proof_type(order.request.requirements.selector)
+            .context("unsupported selector")?
+        {
+            ProofType::Any | ProofType::Inclusion => Ok(base),
+            ProofType::Groth16 => Ok(base + groth16),
+            proof_type => {
+                tracing::warn!("Unknown proof type in gas cost estimation: {proof_type:?}");
+                Ok(base)
+            }
+        }
+    }
```

Comment:
We also need to account for callbacks

Reply:
I've added gas equal to the callback limit